From 2c20281487f68001f4db8f0e3d6316b0df414daf Mon Sep 17 00:00:00 2001
From: Raine Makelainen <raine.makelainen@jolla.com>
Date: Thu, 10 Jan 2019 15:21:24 +0200
Subject: [PATCH] Make fullscreen enabling work as used to with pref
 full-screen-api.content-only. Fixes JB#44129

We don't have chrome from doc shell point of view. This commit
sha1 3116f3bf53df offends fullscreen API to work without chrome
and shall not make root docShell act as chrome. We previously had
"full-screen-api.content-only" pref set to "true".

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 dom/base/nsGlobalWindow.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/dom/base/nsGlobalWindow.cpp b/dom/base/nsGlobalWindow.cpp
index 8ff4b84ce375..2a330a788b0a 100644
--- a/dom/base/nsGlobalWindow.cpp
+++ b/dom/base/nsGlobalWindow.cpp
@@ -6745,10 +6745,16 @@ nsGlobalWindow::SetFullscreenInternal(FullscreenReason aReason,
   if (rootItem != mDocShell)
     return window->SetFullscreenInternal(aReason, aFullScreen);
 
+  // We don't have chrome from doc shell point of view. This commit
+  // sha1 3116f3bf53df offends fullscreen API to work without chrome
+  // and shall not make root docShell act as chrome. We previously had
+  // "full-screen-api.content-only" pref set to "true".
+#if 0
   // make sure we don't try to set full screen on a non-chrome window,
   // which might happen in embedding world
   if (mDocShell->ItemType() != nsIDocShellTreeItem::typeChrome)
     return NS_ERROR_FAILURE;
+#endif
 
   // If we are already in full screen mode, just return.
   if (mFullScreen == aFullScreen)
-- 
2.17.1

